### ISSUE TITLE
error when using .ts file

### ISSUE THREAD

**Issue by GrayStrider** [plopjs/plop#184]
![image](https://user-images.githubusercontent.com/43771776/68070251-3f5b3180-fdc0-11e9-99f2-db8ce99e5ff1.png)
works with .js, what do I do wrong? Am I supposed to compile it myself?

**Comment by GrayStrider** [comment]
compiling manually does nothing

**Comment by amwmedia** [comment]
@crutchcorn @cspotcode thoughts on this?

**Comment by cspotcode** [comment]
@GrayStrider please post a copy of a `plopfile.ts` that reproduces this bug.

**Comment by cspotcode** [comment]
Pretty sure this is cuz plop doesn't support ECMAScript modules.  Or you could say it doesn't support the CommonJS equivalent of ESM.  There's no such thing as replacing the `exports` object in ESM.  It's a flat namespace of exports, where the default export is simply a named export whose name is `default`.  Plop doesn't check for that.

Solution is for plop to use logic like this:

```
function getDefaultExportFunction(requiredExportsFromPlopfile) {
    return typeof requiredExportsFromPlopfile === 'function' ? requiredExportsFromPlopfile : requiredExportsFromPlopfile.default;
    }
}
```

You could also google for what the community does with `__esModule`; maybe someone has strong opinions.  But the above should do the trick.

TypeScript *can* do export re-assignment, but that's not what plop's readme recommends.  So IMO it makes the most sense to use the above logic.

**Comment by crutchcorn** [comment]
This should now be fixed as-of `2.5.4`:

https://github.com/plopjs/plop/releases/tag/v2.5.4

Thanks so much for the PR @zonzujiro
